napari/_qt/containers/_layer_delegate.py

```diff
@@ -92,6 +92,8 @@ def paint(
     def get_layer_icon(self, option, index):
         """Add the appropriate QIcon to the item based on the layer type."""
         layer = index.data(ItemRole)
+        if layer is None:
+            return
```

Comment:
Added this.

My main concern is if we respect the API of QAbstractModelItem:

https://doc.qt.io/qtforpython-5/PySide2/QtCore/QAbstractItemModel.html#PySide2.QtCore.PySide2.QtCore.QAbstractItemModel.data

Reply:
are you referring to the invalid QVariant return type?  If so, `None` is the invalid QVariant for python bindings:
https://doc.qt.io/archives/qtforpython-5.12/pysideapi2.html#qvariant

Reply:
Ah, yes, perfect then.

Reply:
thanks for adding this bit!